packages/smooth_app/lib/database/dao_product.dart

```diff
@@ -130,4 +154,29 @@ class DaoProduct extends AbstractSqlDao implements BulkDeletable {
         json.decode(encodedJson) as Map<String, dynamic>;
     return Product.fromJson(decodedJson);
   }
+
+  Future<String> getFreshLocalDataBase() async {
+    try {
+      final List<String> barcodes = await getAllKeys();
+      if (barcodes.isEmpty) {
+        return 'List is Empty\nNothing to Refresh';
+      }
+      final ProductListQueryConfiguration configuration =
+          ProductListQueryConfiguration(
+        barcodes,
+        fields: ProductQuery.fields,
+        language: ProductQuery.getLanguage(),
+        country: ProductQuery.getCountry(),
+      );
+      final User user = ProductQuery.getUser();
+      // TODO(ashaman999): find a better way to do this not sure at what length it will break
+      final SearchResult products =
+          await OpenFoodAPIClient.getProductList(user, configuration);
+      final List<Product>? productList = products.products;
+      await putAll(productList!);
```

Comment:
Havn't had a close look at the db part but wasn't there a bulk insert

Reply:
putAll  already calls the bulk replace function 

``` dart
 Future<void> putAll(final Iterable<Product> products) async =>
      localDatabase.database.transaction(
        (final Transaction transaction) async =>
            _bulkReplaceLoop(transaction, products),
      );
      
```